evalml/utils/woodwork_utils.py

```diff
@@ -8,68 +8,75 @@
+    return _numpy_to_pandas(np.array(list))
+
+
+_nullable_types = {'Int64', 'Float64', 'boolean'}
+
+
+def _raise_value_error_if_nullable_types_detected(data):
+    types = {data.name: data.dtype} if isinstance(data, pd.Series) else data.dtypes
+    cols_with_nullable_types = {col: str(ptype) for col, ptype in dict(types).items() if str(ptype) in _nullable_types}
+    if cols_with_nullable_types:
+        raise ValueError("Evalml does not support the new pandas nullable types because "
+                         "our dependencies (sklearn, xgboost, lightgbm) do not support them yet."
+                         "If your data does not have missing values, please use the non-nullable types (bool, int64, float64). "
+                         "If your data does have missing values, use float64 for int and float columns and category for boolean columns. "
+                         f"These are the columns with nullable types: {list(cols_with_nullable_types.items())}")
+
+
 def infer_feature_types(data, feature_types=None):
     """Create a Woodwork structure from the given list, pandas, or numpy input, with specified types for columns.
         If a column's type is not specified, it will be inferred by Woodwork.
 
     Arguments:
-        data (pd.DataFrame): Input data to convert to a Woodwork data structure.
+        data (pd.DataFrame, pd.Series): Input data to convert to a Woodwork data structure.
         feature_types (string, ww.logical_type obj, dict, optional): If data is a 2D structure, feature_types must be a dictionary
             mapping column names to the type of data represented in the column. If data is a 1D structure, then feature_types must be
             a Woodwork logical type or a string representing a Woodwork logical type ("Double", "Integer", "Boolean", "Categorical", "Datetime", "NaturalLanguage")
 
     Returns:
         A Woodwork data structure where the data type of each column was either specified or inferred.
     """
-    ww_data = data
-    if isinstance(data, ww.DataTable) or isinstance(data, ww.DataColumn):
-        return ww_data
     if isinstance(data, list):
-        ww_data = np.array(data)
+        data = _list_to_pandas(data)
+    elif isinstance(data, np.ndarray):
+        data = _numpy_to_pandas(data)
 
-    ww_data = ww_data.copy()
-    if len(ww_data.shape) == 1:
-        name = ww_data.name if isinstance(ww_data, pd.Series) else None
-        return ww.DataColumn(ww_data, name=name, logical_type=feature_types)
-    return ww.DataTable(ww_data, logical_types=feature_types)
+    ww_data = data.copy()
 
+    _raise_value_error_if_nullable_types_detected(data)
 
-def _convert_woodwork_types_wrapper(pd_data):
-    """
-    Converts a pandas data structure that may have extension or nullable dtypes to dtypes that numpy can understand and handle.
+    if isinstance(data, pd.Series):
+        if data.ww.schema is not None:
+            ww_data = ww.init_series(ww_data, logical_type=data.ww.logical_type,
```

Comment:
Can't use `series.ww.init(..)` because is the expectation is that `infer_feature_types` should change the logical type if needed. In order to do that with a series, you need to use `ww.init_series`.

https://woodwork.alteryx.com/en/stable/start.html#Using-Woodwork-with-a-Series